lib/route_translator/translator/path/segment.rb

```diff
@@ -5,15 +5,29 @@ module Segment
         class << self
           private
 
-          def translate_string(str, locale)
-            locale = locale.to_s.gsub('native_', '')
-            opts = { scope: :routes, locale: locale }
+          def fallback_options(str, locale)
             if RouteTranslator.config.disable_fallback && locale.to_s != I18n.default_locale.to_s
-              opts[:fallback] = true
+              { fallback: true }
             else
-              opts[:default] = str
+              { default: str }
             end
+          end
+
+          def translate_string(str, locale, scope)
```

Comment:
I still don't like this method, I would avoid the double assignment of res and pots